sdk/cosmosdb/cosmos/src/queryExecutionContext/EndpointComponent/NonStreamingOrderByEndpointComponent.ts

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { DiagnosticNodeInternal } from "../../diagnostics/DiagnosticNodeInternal";
+import { Response } from "../../request";
+import { ExecutionContext } from "../ExecutionContext";
+import { OrderByComparator } from "../orderByComparator";
+import { NonStreamingOrderByResult } from "../nonStreamingOrderByResult";
+import { NonStreamingOrderByPriorityQueue } from "../../utils/nonStreamingOrderByPriorityQueue";
+import { getInitialHeader } from "../headerUtils";
+export class NonStreamingOrderByEndpointComponent implements ExecutionContext {
+  private nonStreamingOrderByPQ: NonStreamingOrderByPriorityQueue<NonStreamingOrderByResult>;
+  private isCompleted: boolean = false;
+  /**
+   * Represents an endpoint in handling an non-streaming order by query. For each processed orderby
+   * result it returns 'payload' item of the result
+   *
+   * @param executionContext - Underlying Execution Context
+   * @hidden
+   */
+  constructor(
+    private executionContext: ExecutionContext,
+    private sortOrders: any[],
+    private priorityQueueBufferSize: number = 2000,
+    private offset: number = 0,
+  ) {
+    const comparator = new OrderByComparator(this.sortOrders);
+    this.nonStreamingOrderByPQ = new NonStreamingOrderByPriorityQueue<NonStreamingOrderByResult>(
+      (a: NonStreamingOrderByResult, b: NonStreamingOrderByResult) => {
+        return comparator.compareItems(b, a);
+      },
+      this.priorityQueueBufferSize,
+    );
+  }
+
+  public async nextItem(diagnosticNode: DiagnosticNodeInternal): Promise<Response<any>> {
+    if (
+      this.priorityQueueBufferSize <= 0 ||
+      (this.isCompleted && this.nonStreamingOrderByPQ.isEmpty())
+    ) {
+      return {
+        result: undefined,
+        headers: getInitialHeader(),
+      };
+    }
+
+    if (this.isCompleted && !this.nonStreamingOrderByPQ.isEmpty()) {
+      const item = this.nonStreamingOrderByPQ.dequeue()?.payload;
+      return {
+        result: item,
+        headers: getInitialHeader(),
+      };
+    }
+    if (this.executionContext.hasMoreResults()) {
+      const { result: item, headers } = await this.executionContext.nextItem(diagnosticNode);
+      if (item !== undefined) {
+        this.nonStreamingOrderByPQ.enqueue(item);
+      }
+      return {
+        result: {},
```

Comment:
Should this be undefined? Also please document the working of this class, Is it that calls to nextItem populate items in priorityQueue, upon some condition `isCompleted` is set true, after that items are returned from queue. May be breaking this logic into functions corresponding to these stages would help. something like -
1. fillPriorityQueue
2. consumePriorityQueue

Reply:
Refactored this. {} is used to signal that there are more results in the backend. undefined is used to indicate no more results in the backend.

Reply:
Added comments to document this behavior.